src/main/java/com/ss/mqtt/broker/handler/packet/in/SubscribeInPacketHandler.java

```diff
@@ -15,5 +19,14 @@
     protected void handleImpl(@NotNull UnsafeMqttClient client, @NotNull SubscribeInPacket packet) {
         var ackReasonCodes = subscriptionService.subscribe(client, packet.getTopicFilters());
         client.send(client.getPacketOutFactory().newSubscribeAck(packet.getPacketId(), ackReasonCodes));
+        ackReasonCodes.forEach(client, SubscribeInPacketHandler::checkException);
+    }
+
+    private static void checkException(@NotNull UnsafeMqttClient client, @NotNull SubscribeAckReasonCode reason) {
+        if (reason == SHARED_SUBSCRIPTIONS_NOT_SUPPORTED || reason == WILDCARD_SUBSCRIPTIONS_NOT_SUPPORTED) {
+            int reasonIndex = Byte.toUnsignedInt(reason.getValue());
+            var disconnect = client.getPacketOutFactory().newDisconnect(client, DisconnectReasonCode.of(reasonIndex));
+            client.sendWithFeedback(disconnect).thenAccept(result -> client.getConnection().close());
```

Comment:
why do you use close instead of release?

Reply:
Because of deduplication